swarm/network/stream/peer.go

```diff
@@ -420,3 +424,132 @@ func (p *Peer) close() {
 		s.Close()
 	}
 }
+
+func (p *Peer) runUpdateSyncing() error {
+	timer := time.NewTimer(p.streamer.syncUpdateDelay)
```

Comment:
The purpose is to avoid very frequent updates when multiple peers are added added or removed in short period of time, like on swarm start or when there are connectivity issues. I think that this delay can be lowered, but I am still for keeping it.